pkg/kwok/metrics/cel/evaluate.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cel provides libraries and evaluators to evaluate metric expressions
+package cel
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Evaluator is a common interface for all types
+// that evaluate metrics
+type Evaluator interface {
+	Evaluate(exp string) (float64, error)
+}
+
+// NewNodeEvaluator returns a MetricEvaluator that is able to evaluate node metrics
+func NewNodeEvaluator(node *corev1.Node) (Evaluator, error) {
+	b, err := json.Marshal(node)
+	if err != nil {
+		return nil, err
+	}
+	m := map[string]any{}
```

Comment:
Is there a better way I can leverage proto buffers instead of json? I had problems doing it, so I just wanted to have something, and then improve it later :smile: